pkg/operator/controller/ingress/status.go

```diff
@@ -979,7 +979,14 @@ func computeDNSStatus(ic *operatorv1.IngressController, wildcardRecord *iov1.DNS
 		}
 	}
 	var conditions []operatorv1.OperatorCondition
-	if ic.Status.EndpointPublishingStrategy.LoadBalancer.DNSManagementPolicy == operatorv1.UnmanagedLoadBalancerDNS {
+	if !manageDNSForDomain(ic.Status.Domain, status, dnsConfig) && ic.Status.EndpointPublishingStrategy.LoadBalancer.DNSManagementPolicy == operatorv1.UnmanagedLoadBalancerDNS {
```

Comment:
What if someone wants it to be unmanaged, and set it that way?  Is there any reason to set this condition in that case?

Reply:
Doesn't this mean that the domains don't match and it is unmanaged?

> 	if !manageDNSForDomain(ic.Status.Domain, status, dnsConfig) && ic.Status.EndpointPublishingStrategy.LoadBalancer.DNSManagementPolicy == operatorv1.UnmanagedLoadBalancerDNS {

Reply:
In other words, if you rely on the mismatch between `domain` and `baseDomain` to determine whether `UnmanagedLoadBalancerDNS` was intentional or not, that is incorrect because the spec domains are not cleaned up when it is intentionally set to unmanaged.  What you would need to do is check the spec against the status DNSManagementPolicy, and if they aren't the same, and there is a mismatch, then you could add this condition.